codec-http2/src/main/java/io/netty/handler/codec/http2/ReadOnlyHttp2Headers.java

```diff
@@ -769,6 +771,30 @@ public CharSequence status() {
         return get(PseudoHeaderName.STATUS.value());
     }
 
+    @Override
+    public boolean contains(CharSequence name, CharSequence value, boolean caseInsensitive) {
+        final int nameHash = AsciiString.hashCode(name);
+        final HashingStrategy<CharSequence> strategy = caseInsensitive? CASE_INSENSITIVE_HASHER : CASE_SENSITIVE_HASHER;
+        final int valueHash = strategy.hashCode(value);
+
+        return contains(name, nameHash, value, valueHash, strategy, pseudoHeaders)
```

Comment:
This should be in the reverse order.   Pseudo headers are checked more rarely, so they should be checked last.

Reply:
sgtm ... can you submit a followup PR?

Reply:
Done, see #7642